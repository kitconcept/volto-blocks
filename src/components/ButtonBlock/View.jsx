import React from 'react';
import { Button } from 'semantic-ui-react';
import cx from 'classnames';
import { ConditionalLink } from '@plone/volto/components';
import { flattenToAppURL, isInternalURL } from '@plone/volto/helpers';

const View = ({ data, isEditMode }) => {
  let link = data?.href ? (
    isInternalURL(data.href) ? (
      <ConditionalLink to={flattenToAppURL(data.href)} condition={!isEditMode}>
        <Button className={(cx('button'), data.align)}>{data.title}</Button>
      </ConditionalLink>
    ) : (
      <a href={flattenToAppURL(data.href)}>
        <Button className={(cx('button'), data.align)}>{data.title}</Button>
      </a>
    )
  ) : (
    <Button className={cx({ noLink: !data.href })}>{data.title}</Button>
  );

  return <div className={cx(`block button align ${data.align}`)}>{link}</div>;
};

export default View;
