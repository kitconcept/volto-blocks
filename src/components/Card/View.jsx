/**
 * View image tile.
 * @module components/manage/Tiles/Hero/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { flattenToAppURL } from '@plone/volto/helpers';
import cx from 'classnames';
import redraft from 'redraft';
import { settings } from '~/config';

/**
 * View image tile class.
 * @class View
 * @extends Component
 */
const View = ({ data, detached }) => (
  <div
    className={cx('tile __card', {
      detached,
    })}
  >
    {(() => {
      const card = (
        <>
          {data.url && (
            <img
              src={`${flattenToAppURL(data.url)}/@@images/image`}
              alt={data.alt || ''}
              className="card-image"
            />
          )}
          <div className="card-title">
            {data.text && (
              <h2>
                {redraft(
                  data.text,
                  settings.ToHTMLRenderers,
                  settings.ToHTMLOptions,
                )}
              </h2>
            )}
          </div>
        </>
      );
      if (data.external) {
        const isReallyExternal =
          (data.external.startsWith('http') ||
            data.external.startsWith('https')) &&
          !data.external.includes(settings.apiPath);

        if (isReallyExternal) {
          return <a href={data.external}>{card}</a>;
        } else {
          return (
            <Link to={data.external.replace(settings.apiPath, '')}>{card}</Link>
          );
        }
      } else if (data.href) {
        return <Link to={data.href}>{card}</Link>;
      } else {
        return card;
      }
    })()}
  </div>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
