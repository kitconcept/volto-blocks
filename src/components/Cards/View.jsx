/**
 * View image tile.
 * @module components/manage/Tiles/Slider/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Card, Image } from 'semantic-ui-react';
import redraft from 'redraft';
import cx from 'classnames';
import { settings } from '~/config';
import { flattenToAppURL } from '@plone/volto/helpers';

const getCardsLenght = cards =>
  cards.length + cards.filter(item => item.x2).length;

/**
 * View image tile class.
 * @class View
 * @extends Component
 */
const View = ({ data }) => (
  <div
    className={cx('tile cards', {
      centered: data.align === 'center' || data.align === undefined,
      'space-between': data.align === 'space-between',
      'centered-text': data.centeredText,
      shrinked: data['x.5'],
    })}
  >
    <Card.Group
      className={cx({
        centered: data.align === 'center' || data.align === undefined,
        'space-between': data.align === 'space-between',
      })}
      itemsPerRow={data.expandCards ? getCardsLenght(data.cards) : 4}
    >
      {data.cards.map(card => (
        <Card
          key={card.id}
          as={card.href ? Link : null}
          className={cx({
            'no-borders': data.noBorders,
            x2: card['x2'],
          })}
          to={card.href || null}
          alt={card.alt || null}
        >
          <Image
            src={
              card.url.startsWith(settings.apiPath)
                ? `${flattenToAppURL(card.url)}/@@images/image`
                : card.url
            }
          />
          {!data.hideText && (
            <Card.Content>
              {card.text &&
                redraft(
                  card.text,
                  settings.ToHTMLRenderers,
                  settings.ToHTMLOptions,
                )}
            </Card.Content>
          )}
        </Card>
      ))}
    </Card.Group>
  </div>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
