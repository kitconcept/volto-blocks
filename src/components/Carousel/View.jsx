import React from 'react';

const CarouselView = (props) => {
  return <></>;
};

export default CarouselView;
