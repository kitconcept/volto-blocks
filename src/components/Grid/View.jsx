import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';
import cx from 'classnames';
import { BlockWrapperEnhancer } from '@kitconcept/volto-blocks/components';

const View = ({ data, render, path }) => {
  return (
    <BlockWrapperEnhancer data={data}>
      <div
        className={cx('block __grid', {
          [data['@type']]: true,
          [data.variation]: data.variation,
          centered: data.align === 'center' || data.align === undefined,
          'space-between': data.align === 'space-between',
          'centered-text': data.centeredText,
          one: data.columns.length === 1,
          two: data.columns.length === 2,
          three: data.columns.length === 3,
          four: data.columns.length === 4,
        })}
      >
        <Grid stackable columns={data.columns.length}>
          {data.columns.map((column) => (
            <Grid.Column key={column.id}>
              {render({ ...column, block: column.id }, path, data.columns)}
            </Grid.Column>
          ))}
        </Grid>
      </div>
    </BlockWrapperEnhancer>
  );
};

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
  render: PropTypes.func.isRequired,
};

export default View;
