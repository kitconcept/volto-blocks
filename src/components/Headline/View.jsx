import React from 'react';

const View = (props) => {
  const { data } = props;
  return (
    <>
      {data && (
        <div className="block headline">
          <div className="wrapper">
            <div className="headline-wrapper">
              <h2>{data?.title}</h2>
            </div>
          </div>
        </div>
      )}
    </>
  );
};

export default View;
