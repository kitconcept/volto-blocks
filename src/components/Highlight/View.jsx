import React from 'react';
import HighlightBody from './HighlightBody';

const View = ({ data, id, properties }) => {
  return (
    <div className="block highlight">
      <HighlightBody data={data} properties={properties} id={id} />
    </div>
  );
};

export default View;
