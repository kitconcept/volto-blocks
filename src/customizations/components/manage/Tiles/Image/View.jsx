/**
 * View image tile.
 * @module components/manage/Tiles/Image/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cx from 'classnames';
import { settings } from '~/config';
import ImageZoom from 'react-medium-image-zoom';

import { flattenToAppURL } from '@plone/volto/helpers';

/**
 * View image tile class.
 * @class View
 * @extends Component
 */
const View = ({ data, detached }) => (
  <p
    className={cx(
      'tile image align',
      {
        center: !Boolean(data.align),
        detached,
      },
      data.align,
    )}
  >
    {data.url && (
      <>
        {(() => {
          const image = (
            <img
              src={
                data.url.includes(settings.apiPath)
                  ? `${flattenToAppURL(data.url)}/@@images/image`
                  : data.url
              }
              alt={data.alt || ''}
            />
          );
          if (data.external) {
            const isReallyExternal =
              (data.external.startsWith('http') ||
                data.external.startsWith('https')) &&
              !data.external.includes(settings.apiPath);

            if (isReallyExternal) {
              return <a href={data.external}>{image}</a>;
            } else {
              return (
                <Link to={data.external.replace(settings.apiPath, '')}>
                  {image}
                </Link>
              );
            }
          } else if (data.href) {
            return <Link to={data.href}>{image}</Link>;
          } else {
            return image;
          }
        })()}
      </>
    )}
  </p>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
