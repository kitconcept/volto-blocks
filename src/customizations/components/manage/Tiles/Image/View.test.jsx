import React from 'react';
import renderer from 'react-test-renderer';
import View from './View';

describe('Image tile View', () => {
  it('renders a view image component', () => {
    const component = renderer.create(<View data={{ url: 'image.jpg' }} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('renders a view image component detached', () => {
    const component = renderer.create(
      <View detached data={{ url: 'image.jpg' }} />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
