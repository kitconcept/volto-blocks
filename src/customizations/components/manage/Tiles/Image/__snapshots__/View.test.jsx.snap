// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image tile View renders a view image component 1`] = `
<p
  className="tile image align center"
>
  <img
    alt=""
    src="image.jpg"
  />
</p>
`;

exports[`Image tile View renders a view image component detached 1`] = `
<p
  className="tile image align center detached"
>
  <img
    alt=""
    src="image.jpg"
  />
</p>
`;
